lib/signed-xml.js

```diff
@@ -344,17 +331,19 @@ SignedXml.prototype.checkSignature = function(xml) {
   this.signingKey = this.keyInfoProvider.getKey(this.keyInfo)
   if (!this.signingKey) throw new Error("key info provider could not resolve key info " + this.keyInfo)
 
-  var doc = new Dom().parseFromString(xml)
+  var doc = xml;
+  if (typeof doc === 'string') {
+    doc = new Dom().parseFromString(xml);
+    this.signedXml = xml;
+  } else {
+    this.signedXml = xml.toString();
+  }
 
   if (!this.validateReferences(doc)) {
     return false;
   }
-  
-  if (!this.validateSignatureValue(doc)) {
-    return false;
-  }
 
-  return true
+  return await this.validateSignatureValue(doc);
```

Comment:
You don't strictly need the `await` here do you? Does it cause an extra entry on the micro task queue?